src/cart/crt_hg.c

```diff
@@ -731,42 +734,45 @@ crt_get_info_string(bool primary, crt_provider_t provider, int iface_idx,
 		D_GOTO(out, rc);
 	}
 
+	/* treat not set and set to empty as the same */
+	no_iface  = (iface_str == NULL || *iface_str == '\0') ? true : false;
+	no_domain = (domain_str == NULL || *domain_str == '\0') ? true : false;
+
```

Comment:
I think we'll have to simplify that, I don't really like that we have all this if (no_iface) and if (no_domain) etc :) it's just hard to understand.

Reply:
sure. some of a mess here is due to us having to form a string from 4 parts, 3 of which are optional. It would have been simpler if we could just fill in 4 fields (provider, domain, interface, port) in a structure as an alternate way of specifying info to mercury:) i feel most providers map well to this 4, and for outliers like 'sm' users can still use string version.